packages/sdk/src/browser-wallets/leather/__tests__/index.spec.ts

```diff
@@ -0,0 +1,388 @@
+      const mockData = {
+        jsonrpc: "2.0",
+        id: "20b0fcf3-ad9a-4f32-8d7f-b6db6fb96207",
+        result: {
+          addresses: [
+            {
+              symbol: "BTC",
+              type: "p2wpkh",
+              address: "tb1qy89jvaya2dzfuagxl7h59pytjs74kxudslux25",
+              publicKey:
+                "0238427342c868536e9c181caf674d242ab8a3e381ee36692ba375e0b315234180",
+              derivationPath: "m/84'/1'/0'/0/0",
+            },
+            {
+              symbol: "BTC",
+              type: "p2tr",
+              address:
+                "tb1pgftv8z8re72kttnyt85k2ae6andngar4c59hasv2kp04kgmmgnnqmxg8uy",
+              publicKey:
+                "023ded01d3e800b07278c5bbc5deee1f3493ebd599ae843936d81a42625a5cfb84",
+              tweakedPublicKey:
+                "3ded01d3e800b07278c5bbc5deee1f3493ebd599ae843936d81a42625a5cfb84",
+              derivationPath: "m/86'/1'/0'/0/0",
+            },
+            {
+              symbol: "STX",
+              address: "ST3TJ35J2T3NCAD471DCN2MB0502Z2TVNPZHN1NEG",
+            },
+          ],
+        },
+      };
+      const network = "mainnet";
+
+      vi.stubGlobal("LeatherProvider", {
+        request: vi.fn().mockResolvedValue(mockData),
+      });
+      expect(getAddresses(network)).rejects.toThrowError(
+        NETWORK_MISMATCH_ERROR,
+      );
+    });
+  });
+
+  describe("signMessage", () => {
+    test("should sign a message", async () => {
+      const mockData = {
+        jsonrpc: "2.0",
+        id: "7930d67d-ecee-44dd-89dc-5159e9f38641",
+        result: {
+          signature:
+            "AUAh7i4gHHEhiKRutcRPX+l3tzpuATA/tVymWh/2x8++pD/w0wm9BD/ORiudYYYsXAuyPVh5FnM4ESnUrDLgdA9D",
+          address:
+            "bc1pl5snnt0zwyet5sxufw8qjglf68luqzjqg5wx44dx978hvsn9kc2svsqfx9",
+          message: "abcdefghijk123456789",
+        },
+      };
+      vi.stubGlobal("LeatherProvider", {
+        request: vi.fn().mockResolvedValue(mockData),
+      });
+      const signedMessageResponse = await signMessage("abcdefghijk123456789", {
+        network: "mainnet",
+        paymentType: "p2tr",
+      });
+
+      expect(signedMessageResponse).toEqual({
+        base64:
+          "AUAh7i4gHHEhiKRutcRPX+l3tzpuATA/tVymWh/2x8++pD/w0wm9BD/ORiudYYYsXAuyPVh5FnM4ESnUrDLgdA9D",
+        hex: "014021ee2e201c712188a46eb5c44f5fe977b73a6e01303fb55ca65a1ff6c7cfbea43ff0d309bd043fce462b9d61862c5c0bb23d58791673381129d4ac32e0740f43",
+      });
+    });
+
+    test("should throw error on user cancel", async () => {
+      const errMessage = "User rejected message signing request";
+      vi.stubGlobal("LeatherProvider", {
+        request: vi.fn().mockImplementation(() => {
+          const error: LeatherErrorResponse = {
+            jsonrpc: "2.0",
+            id: "15327f1a-88a1-4251-a38c-1d4c098ff03f",
+            error: {
+              code: 4001,
+              message: errMessage,
+            },
+          };
+          throw new LeatherError(error);
```

Comment:
`throw error` for Leather as it may not be `instanceof Error`

Reply:
updated too